languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/ApiHttpClient.java

```diff
@@ -1,7 +1,9 @@
 package client;
 
+import java.io.IOException;
+
 public interface ApiHttpClient {
 
-    public ApiHttpResponse execute(final ApiHttpRequest httpRequest);
+    public ApiHttpResponse execute(final ApiHttpRequest httpRequest) throws IOException;
```

Comment:
What is the advantage of this checked exception? What values does it provide for an user of our SDK that he has to catch `IOException`?

Reply:
When you execute http request, IOException might occur, so you have to handle it. OkHttp client throws exception when executing requests, so our ApiHttpClient is also throwing that same exception. The value is that they get to provide additional code to handle the exception if it happens. I mean, same as with every checked exception that exists in Java.

Reply:
And what should they do to handle the `IOException`? I think it's better here not to expose an implementation detail - that OkHttp decided to throw an `IOException` - and that's why I would catch an `IOException` and rethrow it wrapped in a unchecked exception.